airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/metadata.py

```diff
@@ -199,28 +199,28 @@ def valid_metadata_report_dataframe(overrode_metadata_definitions: List[PartialM
 
 
 @asset(group_name=GROUP_NAME)
-def registry_derived_metadata_definitions(
-    cloud_sources_dataframe, cloud_destinations_dataframe, oss_sources_dataframe, oss_destinations_dataframe
+def legacy_registry_derived_metadata_definitions(
+    legacy_cloud_sources_dataframe, legacy_cloud_destinations_dataframe, legacy_oss_sources_dataframe, legacy_oss_destinations_dataframe
 ) -> Output[List[PartialMetadataDefinition]]:
-    sources_metadata_list = merge_into_metadata_definitions("sourceDefinitionId", "source", oss_sources_dataframe, cloud_sources_dataframe)
+    sources_metadata_list = merge_into_metadata_definitions(
+        "sourceDefinitionId", "source", legacy_oss_sources_dataframe, legacy_cloud_sources_dataframe
+    )
     destinations_metadata_list = merge_into_metadata_definitions(
-        "destinationDefinitionId", "destination", oss_destinations_dataframe, cloud_destinations_dataframe
+        "destinationDefinitionId", "destination", legacy_oss_destinations_dataframe, legacy_cloud_destinations_dataframe
     )
     all_definitions = sources_metadata_list + destinations_metadata_list
     return Output(all_definitions, metadata={"count": len(all_definitions)})
 
 
-@asset(required_resource_keys={"metadata_file_blobs"}, group_name=GROUP_NAME)
-def metadata_definitions(context):
-    metadata_file_blobs = context.resources.metadata_file_blobs
+@asset(required_resource_keys={"latest_metadata_file_blobs"}, group_name=GROUP_NAME)
+def metadata_definitions(context: OpExecutionContext) -> List[MetadataDefinition]:
+    latest_metadata_file_blobs = context.resources.latest_metadata_file_blobs
 
     metadata_definitions = []
-    for blob in metadata_file_blobs:
+    for blob in latest_metadata_file_blobs:
         yaml_string = blob.download_as_string().decode("utf-8")
         metadata_dict = yaml.safe_load(yaml_string)
-        metadata_def = ConnectorMetadataDefinitionV1.parse_obj(metadata_dict)
+        metadata_def = MetadataDefinition.parse_obj(metadata_dict)
         metadata_definitions.append(metadata_def)
```

Comment:
nit: Is this "download all metadata files" operation fast enough? Do we have way to download a batch of blobs with the gcloud sdk?